### PR TITLE
Full authentication example

### LINKED ISSUE #24: Add /reset_app and /unauthorized
Based on this discussion on Discord, here's the issue.

> Dennis — 10/12/2024, 15:16
what are /unauthorized and /reset_app used for? they are the path config and mentioned in the ios app code, but nothing for it is set up in the example application. https://github.com/jumpstart-pro/example-hotwire-native-rails-backend/blob/8c1f53e71c770666fb6c181be4359668d3e52825/app/controllers/hotwire/ios/path_configurations_controller.rb#L30-L41

> Joe Masilotti — Yesterday at 18:25
Mind adding an issue to the repo?
>
> /reset_app is used to clear all the tabs and reload the path configuration. It should be redirected to after signing in/out.
>
> /unauthorized happens to catch when a root view, like the first view of a tab, hits a 401 so we can present "You need to sign in to view this tab"

### PR DESCRIPTION
**Pull request by joemasilotti**
This PR adds all the required code for signing in and out via the iOS and Android templates. 

It also fixes some naming (Turbo Native -> Hotwire Native), seeds a user, and cleans up the README.

Closes #24.

### CODE REVIEW

**Comment by stockandawe** [comment]
@joemasilotti , since you are cleaning up the README in this PR, would be good to clarify that this is useful for the Jumpstart Pro Android as well:

```
% git diff README.md                             
diff --git a/README.md b/README.md
index adc00e4..f1a3293 100644
--- a/README.md
+++ b/README.md
@@ -1,6 +1,6 @@
 # Hotwire Native Rails back-end example app
 
-This example app includes everything you need to get up and running with the [Jumpstart Pro iOS](https://jumpstartrails.com/ios) template.
+This example app includes everything you need to get up and running with the [Jumpstart Pro iOS](https://jumpstartrails.com/ios) and/or [Jumpstart Pro Android](https://jumpstartrails.com/android) templates.
 
 You can start with this repository as a template for your Rails codebase or copy/paste the needed code into your existing app.
```

File: app/views/shared/_navbar.html.erb [hunk 1]
```diff
@@ -7,7 +7,13 @@
 
       <div class="sm:ml-6 sm:flex sm:items-center">
         <% if user_signed_in? %>
-          <%= button_to "Sign out", destroy_user_session_path, method: :delete, data: {controller: "turbo-native--sign-out", action: "turbo-native--sign-out#signOut"}, class: "text-gray-900 inline-flex items-center px-3 py-2 text-sm font-medium" %>
+          <%# "Catch" sign out clicks to DELETE to API endpoint for notification token cleanup. %>
+          <%= button_to "Sign out", destroy_user_session_path, method: :delete, data: {
+            controller: "bridge--sign-out",
+            bridge__sign_out_path_value: api_v1_auth_path,
```
Comment (subelsky) [inline app/views/shared/_navbar.html.erb:13]:
```suggestion
            bridge--sign_out_path_value: api_v1_auth_path,
```

Comment (joemasilotti) [reply app/views/shared/_navbar.html.erb:13]: Underscores are converted to dashes when using this Ruby syntax.

**Comment by joemasilotti** [comment]
Good catch @stockandawe. Done!

**Pull request merged**